### PR TITLE
smallvil: Resize Grab

### PR DESCRIPTION
**Pull request by PolyMeilex**

### CODE REVIEW

File: smallvil/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,7 @@ edition = "2021"
 [dependencies]
 slog = { version = "2.1.1" }
 slog-stdlog = "4.1.0"
+bitflags = "*"
```
[the remark this answers is not in the document]
Comment (PolyMeilex) [reply smallvil/Cargo.toml:9]: Oh, that's an overnight

[the remark this answers is not in the document]
Comment (PolyMeilex) [reply smallvil/Cargo.toml]: Done

**Review by Drakulix — APPROVED** [review]
Looks good to me.

I wish we could make the `handle_commit` function be a post-commit hook instead. But because we cannot remove hooks again, this is probably best practice with the current api.

**Comment by HeavyRain266** [comment]
Can we introduce more friendly way to handle input buttons with xkbcommon instead of raw numbers? IMO it feels more beginner friendly. (even tinywl uses xkbcommon)

**Comment by PolyMeilex** [comment]
> Can we introduce more friendly way to handle input buttons with xkbcommon instead of raw numbers? IMO it feels more beginner friendly. (even tinywl uses xkbcommon)

How could xkbcommon be used for that? I don't know any way to convert Linux button codes to xkb. Weston, Sway and Wayfire use `linux/input-event-codes.h` so if we want to remove raw numbers we would use [input_event_codes](https://docs.rs/input-event-codes/latest/input_event_codes/) crate. Is this what you meant?

Is so, then sure, I don't see any downsides of using it.

**Comment by HeavyRain266** [comment]
> How could xkbcommon be used for that? I don't know any way to convert Linux button codes to xkb. Weston, Sway and Wayfire use `linux/input-event-codes.h` so if we want to remove raw numbers we would use [input_event_codes](https://docs.rs/input-event-codes/latest/input_event_codes/) crate. Is this what you meant?
> 
> Is so, then sure, I don't see any downsides of using it.

From smithay you could use e.g. hashbrown::Hashmap (for performance) and then build table with translated raw keycodes to xkb like https://github.com/rust-x-bindings/toy_xcb/blob/230f241324be4ec8765f3d4cf7457e2dbabec941/src/keyboard.rs#L468

**Comment by PolyMeilex** [comment]
Seems unrelated to resize grab per se, so I would just open an issue describing the idea. 
So I'm just merging as is.

**Comment by HeavyRain266** [comment]
> Seems unrelated to resize grab per se, so I would just open an issue describing the idea.
So I'm just merging as is.

First wrote that comment and then realized that those variables with bitflags are for corners and not keyboard buttons... sorry for bothering.

**Pull request merged**